homeassistant/components/home_connect/number.py

```diff
@@ -189,29 +192,49 @@ async def async_set_native_value(self, value: float) -> None:
                 },
             ) from err
 
-    async def async_fetch_constraints(self) -> None:
+    async def async_fetch_constraints(self, _: datetime | None = None) -> None:
         """Fetch the max and min values and step for the number entity."""
-        try:
-            data = await self.coordinator.client.get_setting(
-                self.appliance.info.ha_id, setting_key=SettingKey(self.bsh_key)
-            )
-        except HomeConnectError as err:
-            _LOGGER.error("An error occurred: %s", err)
-        else:
-            self.set_constraints(data)
+        setting_key = cast(SettingKey, self.bsh_key)
+        data = self.appliance.settings.get(setting_key)
+        if not data or not data.unit or not data.constraints:
+            try:
+                data = await self.coordinator.client.get_setting(
+                    self.appliance.info.ha_id, setting_key=setting_key
+                )
+            except TooManyRequestsError as err:
+                async_call_later(
```

Comment:
I think we can make a decorator that handles the exceptions and retries if there's a `TooManyRequestsError`. Then we don't need to copy paste this code around as much.

Reply:
Here's an example of a decorator for an entity method:
https://github.com/home-assistant/core/blob/df006aeaded7e6ba9eba95966a23c237c8d1ce51/homeassistant/components/letpot/entity.py#L43-L64

Reply:
I tried to add the decorator:
https://github.com/home-assistant/core/blob/a770174954bcbe9871bcf3e22243cd7243820907/homeassistant/components/home_connect/entity.py#L132-L170

It enters in the lambda call, but it stops working when is going to call `handler_and_write_ha_state`

Reply:
I don't think that lambda is supported. It could be a functools.partial function.

We need to return in case 1 or 2 below.

https://github.com/home-assistant/core/blob/50ba93042be79abcd0c927abcfe095f1b1e67729/homeassistant/core.py#L378-L391